text/0000-iri-v181-transaction-and-bundle/0000-iri-v181-transaction-and-bundle.md

```diff
@@ -0,0 +1,980 @@
++ Feature name: `iri_v181_transaction_and_bundle`
++ Start date: 2019-09-06
++ RFC PR: [iotaledger/bee-rfcs#3](https://github.com/iotaledger/bee-rfcs/pull/3),
+  [iotaledger/bee-rfcs#18](https://github.com/iotaledger/bee-rfcs/pull/18),
+  [iotaledger/bee-rfcs#20](https://github.com/iotaledger/bee-rfcs/pull/20)
++ Bee issue: [iotaledger/bee#0000](https://github.com/iotaledger/bee/issues/0000)
+
+# Summary
+
+The fundamental communication unit in the IOTA protocol is the transaction. Messages, including payment settlements and
+plain data, are propagated through the IOTA network in transactions. Message that are too large are split up into bundles
+of several transactions and sent one by one.
+
+This RFC proposes a `Transaction` type and a `Bundle` type to represent the transaction and bundle formats used by the
+IOTA Reference Implementation as of release [`iri v1.8.1`]. We also propose the `TransactionBuilder`,
+`IncomingBundleBuilder`, and `OutgoingBundleBuilder` types following the builder pattern to construct both message
+types, respectively. We distinguish between incoming and outgoing bundles because one is concerned with verifying an
+existing message being received, while the other one is concerned with constructing a new message intended to be sent,
+which includes setting a number of fields like the bundle hash.
```

Comment:
I think fields are set in both cases. The differentiating factor is, that for the outgoing bundle we actually have to create/calculate those fields for the first time while for the incoming bundle those fields can just be read from the wire bytes.

Reply:
Reformulated that section.